brayns/common/scene/Scene.cpp

```diff
@@ -147,26 +176,17 @@ void Scene::removeModel(const size_t id)
 {
     {
         std::unique_lock<std::shared_timed_mutex> lock(_modelMutex);
-        auto i = std::find_if(_modelDescriptors.begin(),
-                                _modelDescriptors.end(), [id](auto desc) {
-                                    return id == desc->getModelID();
-                                });
-        if (i == _modelDescriptors.end())
-            return;
-
-        (*i)->callOnRemoved();
-
-        _modelDescriptors.erase(i);
+        auto model =
+            _remove(_modelDescriptors, id, &ModelDescriptor::getModelID);
+        model->callOnRemoved();
```

Comment:
model can be null here so add null check before callOnRemoved().

Reply:
true, good catch